### ISSUE TITLE
[CoE Starter Kit - BUG] Audit Log Subscription Flow fails on start - can't read secret

### ISSUE THREAD

**Issue by kbirstein** [microsoft/coe-starter-kit#6192]
### Does this bug already exist in our backlog?

- [X] I have checked and confirm this is a new bug.

### Describe the issue

While trying to run the Admin | Audit Logs | Office 365 Management API Subscription flow it fails with error: Error occured while reading secret: Value cannot be null.
Parameter name: input

The secret is clearly NOT NULL. I can see it in the Command Center and in the Default Solution environment variable "current" value.

I am not using Azure Key Vault. 

### Expected Behavior

I should be able to run the flow with the "start" parameter successfully.

### What solution are you experiencing the issue with?

Core

### What solution version are you using?

4.11

### What app or flow are you having the issue with?

Admin | Audit Logs | Office 365 Management API Subscription flow 

### What method are you using to get inventory and telemetry?

Cloud flows

### Steps To Reproduce

1. Run the Admin | Audit Logs | Office 365 Management API Subscription flow with the "start" parameter.
2. Flow fails with error 'Error occured while reading secret: Value cannot be null.
Parameter name: input'

### Anything else?

![FlowErrorMessage](https://github.com/microsoft/coe-starter-kit/assets/13750516/5f1b2cd2-08e9-4d4e-9743-2ebe2b4c7942)
![ClientSecret](https://github.com/microsoft/coe-starter-kit/assets/13750516/f21a50de-f675-4e23-8f89-7891b31bc8e4)

**Comment by Jenefer-Monroe** [comment]
Can you please share a screenshot of the API permissions for your app registration?

**Comment by kbirstein** [comment]
![API_Permissions](https://github.com/microsoft/coe-starter-kit/assets/13750516/c3a061c1-93d6-4c02-95dc-d8d935e53f4e)

**Comment by Jenefer-Monroe** [comment]
Thank you, that looks correct.

Please validate that you have entered all the env variables
![image](https://github.com/microsoft/coe-starter-kit/assets/89584709/fec00bdd-5b8d-4a54-8a6e-f72fef04a3a9)

And turn the flows off and back on. Sometimes env vars get stuck in them with old values

**Comment by kbirstein** [comment]
Yes I have entered the admin_AuditLogsClientSecret environment variable. Here's a screenshot.
![AuditLogsClientSecret](https://github.com/microsoft/coe-starter-kit/assets/13750516/d3d340aa-84ed-4f34-9149-0fe37bcd4166)

**Comment by Jenefer-Monroe** [comment]
Sorry I meant all 5 of those. Please validate that you have entered all the env variables
![image](https://github.com/microsoft/coe-starter-kit/assets/89584709/fec00bdd-5b8d-4a54-8a6e-f72fef04a3a9)

Then turn the flow off and on again. 
Also can you confirm, you are a commercial tenant?

**Comment by kbirstein** [comment]
Here you go!
![AuditLogsAuthority](https://github.com/microsoft/coe-starter-kit/assets/13750516/2a4822f8-44bf-4c8e-b6c7-7d1731c15b2f)
![AzureKeyVault](https://github.com/microsoft/coe-starter-kit/assets/13750516/f7789908-0ced-422d-9644-31780fa7ea70)
![ClientID](https://github.com/microsoft/coe-starter-kit/assets/13750516/873be2b1-4b5a-44cf-af7e-d792a8a582d8)
![AuditLogsAudience](https://github.com/microsoft/coe-starter-kit/assets/13750516/e749aead-90b5-43f5-b20b-ab65956c5761)

**Comment by kbirstein** [comment]
Please also note that the account running COE is Global Administrator so there should be no problem with rights.

Should I go back to using the Custom Connector? I'd really like to get this figured out ASAP because I'm installing COE for a client this week!

**Comment by Jenefer-Monroe** [comment]
Did you turn the flow off and on again to see if that fixed it?
Also can you confirm, you are a commercial tenant?

**Comment by kbirstein** [comment]
Ye;s, I am a Commercial tenant and I turned the flow off and on and still the same failure. Maybe I should try creating a new app registration?

**Comment by Jenefer-Monroe** [comment]
I looked at your app registration again and it looks a little different than mine. 
![image](https://github.com/microsoft/coe-starter-kit/assets/89584709/d4573ccf-ac18-4748-b88c-acfca11bebc0)

Maybe just delete and re-add that permission

**Comment by kbirstein** [comment]
I recreated the app registration, put in the new client id and secret, turned the flow off and on and same error. :-(

Note that the instructions say "Select Office 365 Management API" although it's actually has an "s" at the end. . .

![AuditLogInstructions](https://github.com/microsoft/coe-starter-kit/assets/13750516/f8440aa3-c0da-47ac-a5dd-1dc1f5249187)

Here's my new app registration that still doesn't work.

![AppRegistration_COEAuditLog](https://github.com/microsoft/coe-starter-kit/assets/13750516/3a144b70-f4a6-48b8-8498-fc51c013a523)

**Comment by Jenefer-Monroe** [comment]
Open the last attempted run for Admin | Audit Logs | Office 365 Management API Subscription

Is the app id correctly ingested here?
![image](https://github.com/microsoft/coe-starter-kit/assets/89584709/97eee119-aae8-4b4e-95f4-2f5e6bd49feb)

Is the secret correctly ingested here?
![image](https://github.com/microsoft/coe-starter-kit/assets/89584709/9783b8d9-4ddb-40d7-928f-eeeeb9c1145a)

Does this call fail hence correctly marking the variable as false?
![image](https://github.com/microsoft/coe-starter-kit/assets/89584709/0da62240-6cef-42b5-96fd-6b8bea57cb48)

**Comment by kbirstein** [comment]
Hi Jenefer,

The Application (Client) ID is being set properly:

![SetAppIDCurrent](https://github.com/microsoft/coe-starter-kit/assets/13750516/ed8627e8-91bb-480b-8cdd-a17e2a673669)

The Set Azure Secret Type is set properly to "Failed":

![AzureSecretTypeSettoFailed](https://github.com/microsoft/coe-starter-kit/assets/13750516/87916d3d-4cd8-40ac-a897-afea3d0017ff)

The Set Secret value from Text Secret is set to the "Secret ID", not the "Value", which is correct, right?

![SetSecretValueFromTextSecret](https://github.com/microsoft/coe-starter-kit/assets/13750516/3477058e-bbc5-41f5-8a3b-6e0045806404)

Here's the detail error from the Start Subscription action:

![StartErrorinDetail](https://github.com/microsoft/coe-starter-kit/assets/13750516/72432f73-8935-4d76-9b5b-f2477aae8453)

**Comment by Jenefer-Monroe** [comment]
It needs to be the secret, not the ID as shown here. Perhaps you entered it incorrectly at one time, or perhaps you are sharing screenshots from multiple installs and still have the id in one of them?
![image](https://github.com/microsoft/coe-starter-kit/assets/89584709/83279e8b-4070-49f0-8736-909e243873dd)

**Comment by kbirstein** [comment]
HI Jenefer,

I FINALLY figured out the problem. I started checking all the environment variables and realized that somehow I had the wrong value in the Tenant ID variable and the result was that the flow put the Text Secret instead of the Secret Value in the "Set theTextSecret - CurrentValue" action! How WEIRD is that???

So I replaced "admin_TenantID" environment variable with the correct GUID and the flow worked!

Thanks a million for all your help!!!

![WrongTenantID](https://github.com/microsoft/coe-starter-kit/assets/13750516/78f7ad7b-e140-4795-82a0-f472c78b8283)

**Comment by Jenefer-Monroe** [comment]
oh wonderful news. So glad we got this sorted! I'll go ahead and close but please do let us know if you have other issues in the future.

**Comment by MadhanKannathasan** [comment]
Hi Team, 

When we run the flow, we are getting internalservereerror in the start subscription

Encountered internal server error. The tracking I'd is 
3b78c59b-bebf-4622-ac66-cb26a9970ad1

Could you please look in to this issue

**Comment by Jenefer-Monroe** [comment]
> Hi Team,
> 
> When we run the flow, we are getting internalservereerror in the start subscription
> 
> Encountered internal server error. The tracking I'd is 3b78c59b-bebf-4622-ac66-cb26a9970ad1
> 
> Could you please look in to this issue

Please post a new issue for your problem as this one is now closed.

**Comment by RewionAaron** [comment]
Hello everyone, I am currently rolling out the audit logs for a customer in the CoE, but the setup wizard is not running correctly. The Flow “Admin | Audit Logs | Office 365 Management API Subscription” runs on error at the following step. Previously there was a 200 response error indicating an error in our setup. Setup was checked and the Api Client ID, the App Client Secret and the Admin Tenant id were assigned correctly. We fixed the error after we set the App Permissions from Delegated to Application and the Office 365 Management Apis permission was granted to ActivityFeedRead application.

When executing the flow again, we now get the following error after correction:
How do I get this fixed, or at what point do I need to adjust something in the setup to correct the behavior?
We do not use Azure Key Vault but the plaintext option.
 

![Image](https://github.com/user-attachments/assets/88508d3a-8b5e-4ba2-ac1c-504876063be7)
